ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh

```diff
@@ -0,0 +1,201 @@
+        0) subnets="['subnet-030a88e6e97101ab2','subnet-0e07763243186cac5','subnet-02c5fea7482f804fb','subnet-0291499fd1718ee01','subnet-01c4667ad446c8337','subnet-025e9043c44114baa']";;
+        1) subnets="['subnet-0170ee5ccdd7e7823','subnet-0d50cac95bebb5a6e','subnet-0094864467fc2e737','subnet-0daa3919d85296eb6','subnet-0ab1e11d3ed63cc97','subnet-07681ad7ce2b6c281']";;
+        2) subnets="['subnet-00de9462cf29cd3d3','subnet-06595d2851257b4df','subnet-04bbfdd9ca1b67e74','subnet-096992ef7d807f6b4','subnet-0b3d7ba41fc6278b2','subnet-0b99293450e2edb13']";;
+        3) subnets="['subnet-047f6294332aa3c1c','subnet-0c3bce80bbc2c8f1c','subnet-038c38c7d96364d7f','subnet-027a025e9d9db95ce','subnet-04d9008469025b101','subnet-02f75024b00b20a75']";;
+        *) echo >&2 "invalid subnets index"; exit 1;;
+        esac
+        echo "Subnets : ${subnets}"
+    fi
+    cat > "${out}" << EOF
+apiVersion: v1
+baseDomain: ${base_domain}
+metadata:
+  name: ${cluster_name}
+controlPlane:
+  name: master
+  replicas: 3
+  platform:
+    aws:
+      type: ${master_type}
+      zones:
+      - us-east-1a
+      - us-east-1b
+compute:
+- name: worker
+  replicas: ${workers}
+  platform:
+    aws:
+      type: m4.xlarge
+      zones:
+      - us-east-1a
+      - us-east-1b
+platform:
+  aws:
+    region: ${aws_region}
+    userTags:
+      expirationDate: ${expiration_date}
+    subnets: ${subnets}
+pullSecret: >
+  ${pull_secret}
+sshKey: |
+  ${ssh_pub_key}
+EOF
+;;
+azure4)
+    case $((RANDOM % 8)) in
+    0) azure_region=centralus;;
+    1) azure_region=centralus;;
+    2) azure_region=centralus;;
+    3) azure_region=centralus;;
+    4) azure_region=centralus;;
+    5) azure_region=eastus;;
+    6) azure_region=eastus2;;
+    7) azure_region=westus;;
+    esac
+    echo "Azure region: ${azure_region}"
+    vnetrg=""
+    vnetname=""
+    ctrlsubnet=""
+    computesubnet=""
+    if [[ "${cluster_variant}" =~ "shared-vpc" ]]; then
+        vnetrg="os4-common"
+        vnetname="do-not-delete-shared-vnet-${azure_region}"
+        ctrlsubnet="subnet-1"
+        computesubnet="subnet-2"
+    fi
+    cat > "${out}" << EOF
+apiVersion: v1
+baseDomain: ${base_domain}
+metadata:
+  name: ${cluster_name}
+controlPlane:
+  name: master
+  replicas: 3
+compute:
+- name: worker
+  replicas: ${workers}
+platform:
+  azure:
+    baseDomainResourceGroupName: os4-common
+    region: ${azure_region}
+    networkResourceGroupName: ${vnetrg}
+    virtualNetwork: ${vnetname}
+    controlPlaneSubnet: ${ctrlsubnet}
+    computeSubnet: ${computesubnet}
+pullSecret: >
+  ${pull_secret}
+sshKey: |
+  ${ssh_pub_key}
+EOF
+;;
+gcp)
+    gcp_region=us-east1
+    gcp_project=openshift-gce-devel-ci
+    # HACK: try to "poke" the token endpoint before the test starts
+    for i in $(seq 1 30); do
+        code="$( curl -s -o /dev/null -w "%{http_code}" https://oauth2.googleapis.com/token -X POST -d '' || echo "Failed to POST https://oauth2.googleapis.com/token with $?" 1>&2)"
```

Comment:
this poke is not about configuration.  Should this sort of "sanity-test my platform provider" thing happen in its own step?

Reply:
Good as follow-ups when this is in prod.

Reply:
Yeah, breaking this into a separate step in the future is a good idea.